lib/multistream.js

```diff
@@ -103,11 +103,12 @@ function multistream (streamsArray, opts) {
       level = DEFAULT_INFO_LEVEL
     }
 
+    const id = counter++
     const dest_ = {
       stream: stream_,
       level,
       levelVal: undefined,
-      id: counter++
+      id
```

Comment:
Can you explain why this change fixes it?

Reply:
Sorry, it's an unrelated change. At first, I thought it would be useful to listen to the `error` event and automatically remove the stream from the stream list (using the  stream id).

But, as you've mentioned, I think users should listen to the `error` event and decide what to do.

I can revert this change if needed.

The real fix is the `thread-stream` upgrade to 2.0.0.
If you revert this change (i.e., revert thread-stream to 1.x) the added test will fail because it will throw an error.

Does it make sense?

Reply:
> I can revert this change if needed.

Thanks, that'd be awesome.

> The real fix is the thread-stream upgrade to 2.0.0.
If you revert this change (i.e., revert thread-stream to 1.x) the added test will fail because it will throw an error.

Awesome confirmation!

Reply:
Done, thanks for your review!